static/js/message_store.js

```diff
@@ -32,6 +32,12 @@ export function user_ids() {
     return Array.from(message_user_ids);
 }
 
+export function create_mock_message(message) {
+    // For use in tests only. `id` is a required field,
+    // everything else is optional, as required in the test.
+    stored_messages.set(message.id, message);
+}
```

Comment:
I think this function doesn't fully do what I'd had in mind -- the ideal version would populate values that were not provided in some reasonable way such that we end up with a `message` object with all the normal fields it should have.  But I'm OK with merging this for now since this commit improves things by removing overrides.

@showell FYI in case you decide to do one more thing before taking a break from node tests (or want to help 
@abhijeetbodas2001 do such a project).